cranelift-codegen/src/ir/immediates.rs

```diff
@@ -264,6 +265,76 @@ impl FromStr for Uimm32 {
     }
 }
 
+/// A 128-bit unsigned integer immediate operand.
+///
+/// This is used as an immediate value in SIMD instructions
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct Uimm128(pub [u8; 16]);
+
+impl From<u64> for Uimm128 {
+    fn from(x: u64) -> Self {
+        let mut buffer: [u8; 16] = [0; 16]; // zero-fill
+        (0..8).for_each(|byte| buffer[15 - byte] = (x >> (byte as u64 * 8) & 0xff) as u8); // insert each byte from the u64 into v after byte position 8
```

Comment:
Is this little or big endian?

Reply:
I hadn't really thought much about this until you brought it up; thanks! I had always assumed that the `[u8; 16]` would be big-endian. To me that made sense because if I wrote `0x01` in CLIF I would assume that `01` byte is the least significant and rightmost in the array: `[0, 0, ..., 1]`. The code above might be confusing because I am trying to fill in bytes from right-to-left to leave `0`s in the more significant bytes. If I wrote them correctly, the `Uimm128` tests I wrote at the bottom of this file should show big-endianness.

Your comment made me start worrying that I might need the flip the order the bytes are emitted in the `rodata` section. I looked up `MOVAPS` and `MOVUPS` and the Intel manual has these semantics:
```
(V)MOVUPS (128-bit store-form version)
DEST[127:0]  <--  SRC[127:0]
```
So it seems to me that I shouldn't fiddle around with the order too much because in `Uimm128` I don't have to pack/unpack lanes into the 16 bytes; if a user of cranelift wanted to do a `vconst.i32x4` I think they would need to structure their SIMD appropriately for the ISA.